base/opaque_closure.jl

```diff
@@ -0,0 +1,29 @@
+@noinline function (y::Core.OpaqueClosure{A, R})(args...) where {A,R}
+    typeassert(args, A)
+    ccall(y.fptr1, Any, (Any, Ptr{Any}, Int), y, Any[args...], length(args))::R
+end
```

Comment:
This code probably should probably now be moved to C (where `jl_apply_generic` is) as the invoke handler for objects of this type, where we can better deal with race-conditions and updating the pointer after reoptimization and calling convention changes and such (and hopefully optimizations too). I think that'll actually even make the `typeassert` disappear into the dynamic dispatch caching operation on the previous line.